Fix XSS issue on VolunteersJobListingFS page

seanpat09 opened the pull request:
# Critical Changes

# Changes
If External_Signup_Url__c is a javascript action (e.g. `javascript:alert()`), the link will not be rendered on the page.

# Issues Closed
Fixes XSS issues on VolunteersJobListingFS due to External_Signup_Url__c valued used in link

# New Metadata

# Deleted Metadata

seanpat09 commented:
**lurch: attach W-026991

LurchTheButler commented:
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fMM0AAM" target="_blank">W-026991</a>

seanpat09 commented:
@jjbennett I updated the code to use a new field that is conditionally rendered, can you review again? Thanks!

jjbennett commented:
Looks good but a couple questions, did we want to mark the old field deprecated and then also in the changes should we say something to instruct the end user to use the new field for any external sign up urls and that the old field will be removed in a specified amount of time? Under Issues closed there should be an issues number if we do not have an issue to close then just the changes should be good but I think if there is a call to action, it should go under Critical Changes. @sam-knox thoughts?

seanpat09 commented:
@jjbennett I moved the changes to Critical Changes and added that users should start using the new field. I didn't specify a time when it would go away though, I wasn't sure show what's the best timeframe. I also wasn't able to find an Issue for this work item

jjbennett commented:
@seanpat09 - I believe the Issues closed is reserved for issue numbers that we close. This PR and the other both had text there that we should probably remove.

force2b commented:
@seanpat09 and @jjbennett Is this PR going to be merged for the release this sprint or does it need to be held for a future release?

seanpat09 commented:
@force2b I don't think this PR is going to make it into the release for this sprint - we still have to do some testing, documentation and figuring out the best way to depecrate the External_Link_Url__c field

force2b commented:
Thanks @seanpat09. We'll release this sprint without this item. The next release won't be until 2/20.

seanpat09 commented:
@jjbennett added the deprecated label to the field, can you review again?

sam-knox commented:
@seanpat09 - Does it make sense to remove External_Signup_Url__c from the packaged layout if we are deprecating it?

jjbennett commented on src/objects/Volunteer_Job__c.object:
```diff
@@ -82,12 +82,22 @@
         <externalId>false</externalId>
         <inlineHelpText>The external registration site, if any, for this Job. If you include an External Signup URL, V4S redirects to the site from the Sign Up link on the Job Listing FS page.</inlineHelpText>
-        <label>External Signup Url</label>
+        <label>DEPRECATED-External Signup Url</label>
```
@sam-knox - Do you think we should update the help text and description to recommend using the new field as well?

sam-knox replied:
@jjbennett - Yes, good idea. Use this text for both help text and description:

"This field has been deprecated and is no longer in use. Use External Signup Link instead."

seanpat09 commented:
@sam-knox @jjbennett updated the help text and removed External_Signup_Url__c from the packaged layout

seanpat09 commented:
@rponti-sforg This is ready for testing!

davisagli commented:
Since this was reverted I edited the PR text to move the notes above the headings so they won't be included in the release notes.
